packages/core/lib/console.js

```diff
@@ -323,38 +320,47 @@ class Console extends EventEmitter {
       childPath = path.join(__dirname, "../lib/console-child.js");
     }
 
-    // stderr is piped here because we don't need to repeatedly see the parent
-    // errors/warnings in child process - specifically the error re: having
-    // multiple config files
-    const spawnOptions = { stdio: ["inherit", "inherit", "pipe"] };
+    const spawnOptions = { stdio: "pipe" };
     const settings = ["config", "network", "url"]
       .filter(setting => options[setting])
       .map(setting => `--${setting} ${options[setting]}`)
       .join(" ");
 
     const spawnInput = `${settings} -- ${inputStrings}`;
 
-    const spawnResult = spawnSync(
+    const sid = spawn(
       "node",
       ["--no-deprecation", childPath, spawnInput],
       spawnOptions
     );
 
-    if (spawnResult.stderr) {
-      // Theoretically stderr can contain multiple errors.
-      // So let's just print it instead of throwing through
-      // the error handling mechanism. Bad call?
-      debug(spawnResult.stderr.toString());
-    }
+    // Theoretically stderr can contain multiple errors.
+    // So let's jsut print it instaed of throwing through
+    // the error handling mechanism. Bad call?
+    sid.stderr.on("data", data => {
+      debug(data.toString());
+    });
+
+    sid.stdout.on("data", data => {
+      console.log(data.toString().trim());
+    });
 
-    // re-provision to ensure any changes are available in the repl
-    this.provision();
+    return new Promise((resolve, reject) => {
+      sid.on("close", code => {
+        if (!code) {
+          // re-provision to ensure any changes are available in the repl
+          this.provision();
 
-    //display prompt when child repl process is finished
-    this.repl.displayPrompt();
+          //display prompt when child repl process is finished
+          this.repl.displayPrompt();
+          void resolve();
```

Comment:
While `reject` after a `resolve` doesn't do anything, as the Promise will ignore it, it's best to avoid it.

```suggestion
          return void resolve();
```

Reply:
Yikes, thanks!